ArjunaJTA/recovery/src/main/java/org/jboss/narayana/jta/quickstarts/recovery/JmsRecovery.java

```diff
@@ -209,52 +189,42 @@ private void startJTATx(XAResource ... resources) throws XAException, SystemExce
             tm.getTransaction().enlistResource(xaRes);
     }
 
-    private void endJTATx(boolean commit) throws SystemException, RollbackException, HeuristicRollbackException, HeuristicMixedException {
+    private void endJTATx(boolean commit)
+            throws SystemException, RollbackException, HeuristicRollbackException, HeuristicMixedException {
         TransactionManager tm = com.arjuna.ats.jta.TransactionManager.transactionManager();
 
-        // no need to delist resources since that is done automatically by a JTA compliant TM
+        // no need to delist resources since that is done automatically by a JTA
+        // compliant TM
         if (commit)
             tm.commit();
         else
             tm.rollback();
     }
 
-    private void produceMessages(XAConnection connection, String ... msgs) throws Exception{
+    private void produceMessages(XAConnection connection, String... msgs) throws Exception {
         // Create an XA session and a message producer
-        //Session session = connection.createSession(false, Session.AUTO_ACKNOWLEDGE);
-        Session session = connection.createXASession();
-        MessageProducer producer = session.createProducer(queue);
-
-        for (String msg : msgs)
-            producer.send(session.createTextMessage(msg));
-
-        producer.close();
-        session.close();
+        try(Session session = connection.createXASession();
+                MessageProducer producer = session.createProducer(queue);){
+            for (String msg : msgs)
+                producer.send(session.createTextMessage(msg));
+        }
     }
 
-    public void produceMessages(DummyXAResource.faultType fault, String ... messages) throws Exception {
-        XAConnection connection = xacf.createXAConnection();
-
-        try
-        {
+    public void produceMessages(DummyXAResource.faultType fault, String... messages) throws Exception {
+        try(XAConnection connection = xacf.createXAConnection();) {
```

Comment:
Yep, I see your point here. My only concern is whether this is a functional or a style change. I'd lean toward functional here but that's up to you